src/CivicrmEntityViewsData.php

```diff
@@ -382,12 +383,14 @@ protected function processViewsDataForSpecialFields(array &$views_field, $base_t
         break;
 
       case 'civicrm_contact':
+        $user_definition = $this->entityTypeManager->getDefinition('user');
+        assert($user_definition !== NULL);
         $views_field['civicrm_contact']['user'] = [
           'title' => $this->t('User related to the CiviCRM contact'),
           'help' => $this->t('Relate user to the CiviCRM contact.'),
           'relationship' => [
-            'base' => 'users_field_data',
-            'base field' => 'uid',
+            'base' => Database::getConnection()->getFullQualifiedTableName($user_definition->getDataTable()),
+            'base field' => $user_definition->getKey('id'),
```

Comment:
```
Drupal\Core\Database\DatabaseExceptionWrapper: Exception in civicrm_activity view[civicrm_activity_view]: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'db.test87707763users_field_data_civicrm_uf_match.uid' in 'on clause': SELECT "civicrm_activity"."id" AS "id", "civicrm_contact_civicrm_activity_contact"."id" AS "civicrm_contact_civicrm_activity_contact_id" FROM "civicrm_activity" "civicrm_activity" LEFT JOIN "civicrm_activity_contact" "civicrm_activity_contact" ON civicrm_activity.id = civicrm_activity_contact.activity_id AND civicrm_activity_contact.record_type_id = :views_join_condition_0 LEFT JOIN "civicrm_contact" "civicrm_contact_civicrm_activity_contact" ON civicrm_activity_contact.contact_id = civicrm_contact_civicrm_activity_contact.id LEFT JOIN "civicrm_uf_match" "civicrm_contact_civicrm_activity_contact__civicrm_uf_match" ON civicrm_contact_civicrm_activity_contact.id = civicrm_contact_civicrm_activity_contact__civicrm_uf_match.contact_id LEFT JOIN db.test87707763users_field_data "dbtest87707763users_field_data_civicrm_uf_match" ON civicrm_contact_civicrm_activity_contact__civicrm_uf_match.uf_id = db.test87707763users_field_data_civicrm_uf_match.uid LIMIT 11 OFFSET 0; Array ( [:views_join_condition_0] => 2 ) in main() (line 19 of index.php).
```

Kind of close... but civicrm_uf_match gets added

Reply:
````
LEFT JOIN db.test87707763users_field_data "dbtest87707763users_field_data_civicrm_uf_match" 
ON civicrm_contact_civicrm_activity_contact__civicrm_uf_match.uf_id = db.test87707763users_field_data_civicrm_uf_match.uid
```

Okay, so that alias is correct. Kind of. It's adding `db.` to the table alias which already has `db.`

So


* `'db.test87707763users_field_data_civicrm_uf_match.uid'`
* Reallyt ends up being `db.db.test87707763users_field_data`